File: src/main/java/com/cloudforce/service/DashboardService.java
package com.cloudforce.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.cloudforce.domain.Dashboard;
import com.cloudforce.dto.DashboardDTO;
import com.cloudforce.dto.DashboardSearchDTO;
import com.cloudforce.dto.DashboardPageDTO;
import com.cloudforce.dto.DashboardConvertCriteriaDTO;
import com.cloudforce.service.GenericService;
import com.cloudforce.dto.common.RequestDTO;
import com.cloudforce.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface DashboardService extends GenericService<Dashboard, Integer> {

	List<Dashboard> findAll();

	ResultDTO addDashboard(DashboardDTO dashboardDTO, RequestDTO requestDTO);

	ResultDTO updateDashboard(DashboardDTO dashboardDTO, RequestDTO requestDTO);

    Page<Dashboard> getAllDashboards(Pageable pageable);

    Page<Dashboard> getAllDashboards(Specification<Dashboard> spec, Pageable pageable);

	ResponseEntity<DashboardPageDTO> getDashboards(DashboardSearchDTO dashboardSearchDTO);
	
	List<DashboardDTO> convertDashboardsToDashboardDTOs(List<Dashboard> dashboards, DashboardConvertCriteriaDTO convertCriteria);

	DashboardDTO getDashboardDTOById(Integer dashboardId);


}

File: cloudforce_ui/src/services/DashboardService.js
import http from "../http-common"; 

class DashboardService {
  getAllDashboards(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/dashboard/dashboards`, searchDTO);
  }
 

  get(dashboardId) {
    return this.getRequest(`/dashboard/${dashboardId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/dashboard?field=${matchData}`, null);
  }

  addDashboard(data) {
    return http.post("/dashboard/addDashboard", data);
  }

  update(data) {
  	return http.post("/dashboard/updateDashboard", data);
  }
  
  uploadImage(data,dashboardId) {
  	return http.postForm("/dashboard/uploadImage/"+dashboardId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new DashboardService();
